### PR TITLE
Enable ExtraRepositories in subscription

### PR DESCRIPTION
**Pull request by stephdl**
Enable extra repositories for subscription

https://github.com/NethServer/dev/issues/6427

### CODE REVIEW

**Review by gsanchietti — APPROVED** [review]
It seems good to me, but let's wait Davide approval

File: root/etc/e-smith/templates/etc/nethserver/eorepo.conf/20subscription [hunk 1]
```diff
@@ -15,7 +15,11 @@
...
+        if ($subscription{'ExtraRepositories'}) {
+            my @ExtraRepositories = split(',',$subscription{'ExtraRepositories'});
+            push (@repos, @ExtraRepositories);
+        }
     }
 
     '';
-}
\ No newline at end of file
+}
```
Comment (DavidePrincipi) [inline root/etc/e-smith/templates/etc/nethserver/eorepo.conf/20subscription]: Remove EOL at EOF

File: root/etc/e-smith/templates/etc/nethserver/pkginfo.conf/20subscription [hunk 2]
```diff
@@ -10,6 +10,10 @@
...
         );
+        if ($subscription{'ExtraRepositories'}) {
+            my @ExtraRepositories = split(',',$subscription{'ExtraRepositories'});
+            push (@repo_list, @ExtraRepositories);
+        }
     }
     '';
-}
\ No newline at end of file
+}
```
Comment (DavidePrincipi) [inline root/etc/e-smith/templates/etc/nethserver/pkginfo.conf/20subscription]: Remove EOL at EOF

**Review by DavidePrincipi — CHANGES_REQUESTED** [review]
Thank you for this PR!

Please remove the trailing end-of-line characters! The modified template fragments must produce an empty string as output.

Question: is there any reason to consider the `ExtraRepositories` prop only with an enabled subscription? I think it can be useful also for community users as well...

**Comment by stephdl** [comment]
> Question: is there any reason to consider the ExtraRepositories prop only with an enabled subscription? I think it can be useful also for community users as well...

It is the same problem I faced as an owner of repository, make this repo enabled after the event software-repos-save, I do not know, I would prefer that the sys-admin or the developer push a fragment to eorepo, however simplicity is often good too

I would like the advice of @gsanchietti

**Comment by DavidePrincipi** [comment]
> I would prefer that the sys-admin or the developer push a fragment to eorepo, however simplicity is often good too

Yes I agree here. The NethServer developer with a public YUM repo should provide a fragment for eorepo in his `customrepo-release.rpm`. 

However a NethServer user/sysadmin might want to enable a third-party repo, for say a custom kernel module or whatever non-NethServer related addon... In this scenario the new prop would be useful, as it does not require custom-templates.

**Comment by gsanchietti** [comment]
> It is the same problem I faced as an owner of repository, make this repo enabled after the event software-repos-save, I do not know, I would prefer that the sys-admin or the developer push a fragment to eorepo, however simplicity is often good too

We can enable use the prop if if the subscription is not enabled, it's ok for me ;)

File: README.rst [hunk 3]
```diff
@@ -87,6 +88,17 @@ initial state and use the original NethServer repositories run ::
   config setprop subscription SystemId '' Secret ''
   signal-event software-repos-save
 
+Enable extra repositories in subscription
```
Comment (DavidePrincipi) [inline README.rst:91]:
```suggestion
Enable extra repositories
```

**Review by DavidePrincipi — COMMENTED** [review]
I saw the code update however the `ExtraRepositories` seems still subject to an enabled subscription. From the last comment by Giacomo, I was expecting to see it honored unconditionally...

File: README.rst [hunk 4]
```diff
@@ -27,6 +27,7 @@ Properties:
 - ``SiteUrl``: Product website URL
 - ``DocsUrl``: Product documentation URL
 - ``HelpdeskUrl``: Product helpdesk URL
+- ``ExtraRepositories``: Enable extra repositories in the subscription (comma separated : nethforge,repoA,repoB).
```
Comment (DavidePrincipi) [inline README.rst:30]:
```suggestion
- ``ExtraRepositories``: Enable extra repositories (comma separated : nethforge,repoA,repoB).
```

**Review by gsanchietti — CHANGES_REQUESTED** [review]
Please add an empty `ExtraRepositories` prop to db defaults

**Review by gsanchietti — APPROVED** [review]

**Pull request merged**